### PR TITLE
TXIF-2371 - Adding TWTG drivers for TS, VS MT & VS QT

### PR DESCRIPTION
**Pull request by tomHallActility**
Adding TWTG drivers for TS, Valve Sensor MT & QT
Still needs better examples.

### CODE REVIEW

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/examples.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
+[
+    {
+        "description": "Payload for Neon Valve Sensor MT 1",
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/examples.json:3]: YOu have used examples of VS-MT driver on a VS-QT driver.. is it only the description of the example which is wrong? if not, we have to remove these examples from this driver.

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Formal driver name
+name: TWTG Neon Temperature Sensor Driver
+# Simple description of the driver functionalities
+description: A driver that decodes and encodes TWTG Neon Temperature Sensor.
+# Mandatory - id of the company that developed the driver
+# This id will be used to construct the id of the driver (Should be in lowercase, no special characters, maximum 8 characters)
+producerId: twtg
+# Email of the developer
+developerEmail: marten.lootsma@twtg.io
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml:9]: This email should be used like the other twtg drivers: service.devices@twtg.io

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Formal driver name
+name: TWTG Neon Valve Sensor MT Driver
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml:2]: TWTG Neon Multi Turn Valve Sensor Driver

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# Simple description of the driver functionalities
+description: A driver that decodes and encodes TWTG Neon Valve Sensor MT.
+# Mandatory - id of the company that developed the driver
+# This id will be used to construct the id of the driver (Should be in lowercase, no special characters, maximum 8 characters)
+producerId: twtg
+# Email of the developer
+developerEmail: marten.lootsma@twtg.io
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml:9]: service.devices@twtg.io

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "neon-vs-mt-xx-xx",
+  "version": "1.0.0-SNAPSHOT",
+  "description": "NEON Temperature Transmitter driver",
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json:4]: This is not a temperature transmitter driver, it is NEON Multi Turn Valve driver

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Formal driver name
+name: TWTG Neon Valve Sensor QT Driver
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml:2]: TWTG Neon Quater Turn Valve Sensor Driver

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+# Simple description of the driver functionalities
+description: A driver that decodes and encodes TWTG Neon Valve Sensor QT.
+# Mandatory - id of the company that developed the driver
+# This id will be used to construct the id of the driver (Should be in lowercase, no special characters, maximum 8 characters)
+producerId: twtg
+# Email of the developer
+developerEmail: marten.lootsma@twtg.io
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml:9]: service.devices@twtg.io

**Review by mostafaibr — CHANGES_REQUESTED** [review]

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json [hunk 8]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "neon-ds-ts-xx-xx",
+  "version": "1.0.0-SNAPSHOT",
+  "description": "NEON Temperature Transmitter driver",
```
Comment (mostafaibr) [inline vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json:4]: Temperature Sensor not Temperature Transmitter

**Comment by mostafaibr** [comment]
Add protocolId of the drivers on both VS-MT and VS-QT models (on model.yaml)

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# Formal driver name
+name: TWTG Neon Temperature Sensor Driver
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml:2]:
```suggestion
name: TWTG NEON Temperature Sensor driver
```

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+# Simple description of the driver functionalities
+description: A driver that decodes and encodes TWTG Neon Temperature Sensor.
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml:4]:
```suggestion
description: A driver that decodes and encodes TWTG NEON Temperature Sensor.
```

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+# Mandatory - id of the company that developed the driver
+# This id will be used to construct the id of the driver (Should be in lowercase, no special characters, maximum 8 characters)
+producerId: twtg
+# Email of the developer
+developerEmail: marten.lootsma@twtg.io
+# In case of a driver per model: `<vendorId>:<modelId>:<modelVersion>`
+# In case of a driver for multiples model: `<vendorId>:generic:<genericVersion>`
+# Needed for linking the model with a specific driver -> must be the same one used in model.yaml in the corresponding model
+protocolId: twtg:ts:1
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml:13]: To where this 1 corresponds to?

In our binary message we have a protocol version field. Typically our drivers are compatible with all versions in the field. The highest protocol version for the TS is 2. Shouldn't this number be 2 as well? Or is this decoupled? If that is the case, what determines this number?

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json [hunk 12]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "neon-ds-ts-xx-xx",
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json:2]:
```suggestion
  "name": "neon-ts",
```

Same counts for the folder name

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml (hunk 3 above)
Comment (martinichka) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml]:
```suggestion
name: TWTG NEON Multi Turn Valve Sensor driver
```

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml [hunk 13]
```diff
@@ -0,0 +1,16 @@
+# Formal driver name
+name: TWTG Neon Valve Sensor MT Driver
+# Simple description of the driver functionalities
+description: A driver that decodes and encodes TWTG Neon Valve Sensor MT.
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml:4]:
```suggestion
description: A driver that decodes and encodes TWTG NEON Multi Turn Valve Sensor.
```

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+# Mandatory - id of the company that developed the driver
+# This id will be used to construct the id of the driver (Should be in lowercase, no special characters, maximum 8 characters)
+producerId: twtg
+# Email of the developer
+developerEmail: marten.lootsma@twtg.io
+# In case of a driver per model: `<vendorId>:<modelId>:<modelVersion>`
+# In case of a driver for multiples model: `<vendorId>:generic:<genericVersion>`
+# Needed for linking the model with a specific driver -> must be the same one used in model.yaml in the corresponding model
+protocolId: twtg:vs-mt:1
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml:13]: Same question as for `ts`

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json [hunk 15]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "neon-vs-mt-xx-xx",
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json:2]:
```suggestion
  "name": "neon-vs-mt",
```

Same counts for the folder name

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+  "version": "1.0.0-SNAPSHOT",
+  "description": "NEON Multi Turn Valve driver",
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json:4]:
```suggestion
  "description": "NEON Multi Turn Valve Sensor driver",
```

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json (hunk 8 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json]: Corrected

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml (hunk 7 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml]: Corrected

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml (hunk 6 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml]: Corrected

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json (hunk 5 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json]: Corrected

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml (hunk 4 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml]: Corrected

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml (hunk 3 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml]: Corrected

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml (hunk 2 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml]: Corrected

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/examples.json (hunk 1 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-qt-xx-xx/examples.json]: The description was the problem. Examples seem still valid.

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml [hunk 17]
```diff
@@ -0,0 +1,16 @@
+# Formal driver name
+name: TWTG Neon Quarter Turn Valve Sensor Driver
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml:2]:
```suggestion
name: TWTG NEON Quarter Turn Valve Sensor driver
```

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+# Simple description of the driver functionalities
+description: A driver that decodes and encodes TWTG Neon Valve Sensor QT.
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml:4]:
```suggestion
description: A driver that decodes and encodes TWTG NEON Quarter Turn Valve Sensor.
```

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+# Mandatory - id of the company that developed the driver
+# This id will be used to construct the id of the driver (Should be in lowercase, no special characters, maximum 8 characters)
+producerId: twtg
+# Email of the developer
+developerEmail: service.devices@twtg.io
+# In case of a driver per model: `<vendorId>:<modelId>:<modelVersion>`
+# In case of a driver for multiples model: `<vendorId>:generic:<genericVersion>`
+# Needed for linking the model with a specific driver -> must be the same one used in model.yaml in the corresponding model
+protocolId: twtg:vs-qt:1
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml:13]: Same question as for `ts`

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/package.json [hunk 20]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "neon-vs-qt-xx-xx",
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/package.json:2]:
```suggestion
  "name": "neon-vs-qt",
```

Same counts for the folder name

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/package.json [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+  "version": "1.0.0-SNAPSHOT",
+  "description": "NEON Valve Sensor QT driver",
```
Comment (martinichka) [inline vendors/twtg/drivers/neon-vs-qt-xx-xx/package.json:4]:
```suggestion
  "description": "NEON Quarter Turn Valve Sensor driver",
```

File: vendors/twtg/models/neon-vs-qt/model.yaml [hunk 22]
```diff
@@ -31,10 +31,17 @@ LoRaWANCertified: false
 # Always mandatory: `<vendorId>:<modelName>:<modelVersion>` (You should have only one per model)
 modelId: twtg:neon-vs-qt:1
 # <vendorId>:<protocolName>:<protocolVersion> Needed for linking the model with a specific driver -> must be the same one used in driver.yaml in the corresponding model (You might have several ones)
-protocolId:
+protocolId: twtg:vs-qt:1
 # DataSheet URL (optional)
 specificationURL: https://github.com/TWTG-R-D-B-V/neon-product-documentation/blob/main/VS-QT/TWTG-NEON_Valve-sensor-QT_Product-Sheet_B1.pdf
```
Comment (martinichka) [inline vendors/twtg/models/neon-vs-qt/model.yaml:36]: Although it is correct, I prefer to have the link to the parent folder, such that on document version updates we don't need to update the links everywhere
```suggestion
specificationURL: https://github.com/TWTG-R-D-B-V/neon-product-documentation/tree/main/VS-QT
```

File: vendors/twtg/models/neon-vs-mt/model.yaml [hunk 23]
```diff
@@ -31,10 +31,17 @@ LoRaWANCertified: false
 # Always mandatory: `<vendorId>:<modelName>:<modelVersion>` (You should have only one per model)
 modelId: twtg:neon-vs-mt:1
 # <vendorId>:<protocolName>:<protocolVersion> Needed for linking the model with a specific driver -> must be the same one used in driver.yaml in the corresponding model (You might have several ones)
-protocolId:
+protocolId: twtg:vs-mt:1
 # DataSheet URL (optional)
 specificationURL: https://github.com/TWTG-R-D-B-V/neon-product-documentation/blob/main/VS-MT/TWTG%20-NEON_Valve%20-Sensor-MT_Product-Sheet_B1.pdf
```
Comment (martinichka) [inline vendors/twtg/models/neon-vs-mt/model.yaml:36]: Although it is correct, I prefer to have the link to the parent folder, such that on document version updates we don't need to update the links everywhere
```suggestion
specificationURL: https://github.com/TWTG-R-D-B-V/neon-product-documentation/tree/main/VS-MT
```

File: vendors/twtg/models/neon-ts/model.yaml [hunk 24]
```diff
@@ -31,11 +31,20 @@ LoRaWANCertified: false
 # Always mandatory: `<vendorId>:<modelName>:<modelVersion>` (You should have only one per model)
 modelId: twtg:neon-ts:1
 # <vendorId>:<protocolName>:<protocolVersion> Needed for linking the model with a specific driver -> must be the same one used in driver.yaml in the corresponding model (You might have several ones)
-protocolId:
+protocolId: twtg:ts:1
 # DataSheet URL (optional)
 specificationURL: https://github.com/TWTG-R-D-B-V/neon-product-documentation/blob/main/TS/TWTG-NEON_Temperature-Sensor_Product-Sheet_B1.pdf
```
Comment (martinichka) [inline vendors/twtg/models/neon-ts/model.yaml:36]: Although it is correct, I prefer to have the link to the parent folder, such that on document version updates we don't need to update the links everywhere
```suggestion
specificationURL: https://github.com/TWTG-R-D-B-V/neon-product-documentation/tree/main/TT
```

**Review by martinichka — CHANGES_REQUESTED** [review]
Please update the folder names according the suggested driver ID's

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml (hunk 11 above)
Comment (martinichka) [reply vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml:13]: Yes, for all drivers you add in this PR (TS, VS-MT, VS-QT) it is protocol version 2.

For VS-CS (which is not part of this PR) it is protocol version 1

Comment (tomHallActility) [reply vendors/twtg/drivers/neon-ds-ts-xx-xx/driver.yaml]: Corrected on Temperature Sensor, and MT and QT Valve Sensors

File: vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json (hunk 12 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-ds-ts-xx-xx/package.json]: Corrected on Temperature Sensor

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml (hunk 3 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml]: Corrected on MT Valve Sensor

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml (hunk 13 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml]: Corrected on MT Valve Sensor

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml (hunk 14 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/driver.yaml]: Corrected on MT Valve Sensor

File: vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json (hunk 15 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-mt-xx-xx/package.json]: Corrected on MT Valve Sensor

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml (hunk 17 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml]: Corrected on QT Valve Sensor

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml (hunk 18 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml]: Corrected on QT Valve Sensor

File: vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml (hunk 19 above)
Comment (tomHallActility) [reply vendors/twtg/drivers/neon-vs-qt-xx-xx/driver.yaml]: Corrected on QT Valve Sensor

File: vendors/twtg/models/neon-vs-qt/model.yaml (hunk 22 above)
Comment (tomHallActility) [reply vendors/twtg/models/neon-vs-qt/model.yaml]: Corrected on QT Valve Sensor

File: vendors/twtg/models/neon-ts/model.yaml (hunk 24 above)
Comment (tomHallActility) [reply vendors/twtg/models/neon-ts/model.yaml]: Corrected on Temperature Sensor

File: vendors/twtg/models/neon-ts/model.yaml [hunk 25]
```diff
@@ -33,7 +33,7 @@ modelId: twtg:neon-ts:1
 # <vendorId>:<protocolName>:<protocolVersion> Needed for linking the model with a specific driver -> must be the same one used in driver.yaml in the corresponding model (You might have several ones)
 protocolId: twtg:ts:1
```
Comment (mostafaibr) [inline vendors/twtg/models/neon-ts/model.yaml:34]: protocolId should be updated with the new one

File: vendors/twtg/models/neon-vs-qt/model.yaml [hunk 26]
```diff
@@ -33,7 +33,7 @@ modelId: twtg:neon-vs-qt:1
 # <vendorId>:<protocolName>:<protocolVersion> Needed for linking the model with a specific driver -> must be the same one used in driver.yaml in the corresponding model (You might have several ones)
 protocolId: twtg:vs-qt:1
```
Comment (mostafaibr) [inline vendors/twtg/models/neon-vs-qt/model.yaml:34]: protocolId should be updated with the new one

**Review by mostafaibr — CHANGES_REQUESTED** [review]

File: vendors/twtg/models/neon-ts/model.yaml (hunk 25 above)
Comment (tomHallActility) [reply vendors/twtg/models/neon-ts/model.yaml]: Corrected

File: vendors/twtg/models/neon-vs-mt/model.yaml [hunk 27]
```diff
@@ -33,7 +33,7 @@ modelId: twtg:neon-vs-mt:1
 # <vendorId>:<protocolName>:<protocolVersion> Needed for linking the model with a specific driver -> must be the same one used in driver.yaml in the corresponding model (You might have several ones)
 protocolId: twtg:vs-mt:1
```
[the remark this answers is not in the document]
Comment (tomHallActility) [reply vendors/twtg/models/neon-vs-mt/model.yaml]: Corrected

**Review by mostafaibr — APPROVED** [review]

**Pull request merged**